src/ext/UtilExtension/wixext/UtilCompiler.cs

```diff
@@ -2446,13 +2451,15 @@ private void ParsePermissionExElement(XElement node, string objectId, string com
             int permission = 0;
             string[] specialPermissions = null;
             string user = null;
+            int attributes = 0;
 
             PermissionType permissionType = PermissionType.SecureObjects;
 
             switch (tableName)
             {
                 case "CreateFolder":
                     specialPermissions = UtilConstants.FolderPermissions;
+                    attributes |= (int)WixPermissionExAttributes.Inheritable; // For backward compatibility the default handling for folders is to enable inheritance, all others are not enabled
```

Comment:
My vote is to get rid of this.  Yes, this is how v3 worked but I think all types should have the same default.

Note that with this current code, there's no way to opt out of inheritance on `CreateFolder`.